File: 0169_majority_element/majority_element.java
class Solution {
    // Time: O(n)
    // Space: O(1)
    public int majorityElement(int[] nums) {
        int candidate = 0;
        int count = 0;
        for (int num : nums) {
            if (count == 0) {
                candidate = num;
                count++;
            } else if (num == candidate) {
                count++;
            } else {
                count--;
            }
        }
        return candidate;
    }
}

File: 0169_majority_element/majority_element.go
package main

/**
 * Time: O(n)
 * Space: O(1)
 */
func majorityElement(nums []int) int {
	count, candidate := 0, 0
	for _, n := range nums {
		if count == 0 {
			candidate = n
		}
		if n == candidate {
			count++
		} else {
			count--
		}
	}
	return candidate
}

func main() {

}
